File: LeetCode/Easy/0338-counting-bits/0338-counting-bits.py
bits_cnt = [0]


class Solution:
    def countBits(self, n: int) -> List[int]:
        for i in range(len(bits_cnt), n + 1):
            bits_cnt.append(bits_cnt[i // 2] + i % 2)
        return bits_cnt[: n + 1]

File: LeetCode/Easy/0338-counting-bits/0338-counting-bits.rs
impl Solution {
    pub fn count_bits(n: i32) -> Vec<i32> {
        let mut v = vec![0; n as usize + 1];
        for i in 1..=n {
            v[i as usize] = v[i as usize / 2] + i % 2;
        }
        v
    }
}
